File: estDados_python/aula_13_04/principal.py
from matriz import *

def main():
    dimensao = 10

    # Criar uma matriz de dimensão 10x10
    matriz = [[0 for _ in range(dimensao)] for _ in range(dimensao)]

    inicializar_matriz(matriz) # type: ignore

    matriz[0][3] = 1
    matriz[2][5] = 1
    matriz[5][7] = 1
    matriz[6][8] = 1

    exibir_matriz(matriz) # type: ignore

    lista = converter_matriz_lista(matriz) # type: ignore

    exibir_lista(lista) # type: ignore

if __name__ == "__main__":
    main()

File: estDados_Java/aula_13_04/Principal.java
package aula_13_04;

import java.util.ArrayList;

public class Principal {
    public static void main(String[] args) {
        //Simulando uma matriz especial, onde os dados são armazenados em uma matriz de inteiros, onde o valor 1 representa a presença de um dado e o valor 0 representa a ausência de um dado.
        int dimansao = 10;
        int matriz[][] = new int[dimansao][dimansao];
        Matriz.inicializarmatriz(matriz, dimansao, dimansao);

        matriz[0][3] = 1;
        matriz[2][5] = 1;
        matriz[5][7] = 1;
        matriz[6][8] = 1;
        Matriz.exibirMatriz(matriz, dimansao, dimansao);

        //converter a matriz especial em lista de dados
        ArrayList<Dado> lista = new ArrayList<>();
        Matriz.converterMatrizEmLista(matriz, dimansao, dimansao, lista);
        Matriz.exibirLista(lista);
    }
}
